pkg/bpf/detect_linux.go

```diff
@@ -78,8 +76,12 @@ func detectKprobeMulti() bool {
 	syms := []string{"vprintk"}
 	opts := link.KprobeMultiOptions{Symbols: syms}
 
-	_, err = link.KprobeMulti(prog, opts)
-	return err == nil
+	link, err := link.KprobeMulti(prog, opts)
+	if err != nil {
+		return false
+	}
+	link.Close()
+	return true
```

Comment:
```suggestion
	link, err = link.KprobeMulti(prog, opts)
	defer link.Close()
	return err == nil
```

Reply:
BTW, I agree, I just noticed it running a test and looking at the loaded ebpf progs with bpftool, otherwise I would have never seen it

Reply:
> uhm if err!=nil, link would probably be nil, so there is the risk of ending up calling link.Close() on a nil pointer

you are right sorry!